src/main/java/com/shopify/graphql/support/AbstractResponse.java

```diff
@@ -62,6 +69,108 @@ public Object get(String field) {
+    
+    /**
+     * Tries to deserialise and return the given JSON field as an Integer. The field itself is expected
+     * to be stored in the object data as a {@link JsonElement}, which is the case for custom simple fields.
+     * 
+     * @param field The name of the field.
+     * @return The value of the field.
+     */
+    public Integer getAsInteger(String field) throws SchemaViolationError {
+        BiFunction<JsonElement, String, Integer> converter = converterWrapper(this::jsonAsInteger);
+        return getAs(field, converter);
+    }
+
+    /**
+     * Tries to deserialise and return the given JSON field as a Double. The field itself is expected
+     * to be stored in the object data as a {@link JsonElement}, which is the case for custom simple fields.
+     * 
+     * @param field The name of the field.
+     * @return The value of the field.
+     */
+    public Double getAsDouble(String field) throws SchemaViolationError {
+        BiFunction<JsonElement, String, Double> converter = converterWrapper(this::jsonAsDouble);
+        return getAs(field, converter);
+    }
+
+    /**
+     * Tries to deserialise and return the given JSON field as a Boolean. The field itself is expected
+     * to be stored in the object data as a {@link JsonElement}, which is the case for custom simple fields.
+     * 
+     * @param field The name of the field.
+     * @return The value of the field.
+     */
+    public Boolean getAsBoolean(String field) throws SchemaViolationError {
+        BiFunction<JsonElement, String, Boolean> converter = converterWrapper(this::jsonAsBoolean);
+        return getAs(field, converter);
+    }
+
+    /**
+     * Tries to deserialise and return the given JSON field as an Array. The field itself is expected
+     * to be stored in the object data as a {@link JsonElement}, which is the case for custom simple fields.
+     * 
+     * @param field The name of the field.
+     * @return The value of the field.
+     */
+    public JsonArray getAsArray(String field) throws SchemaViolationError {
+        BiFunction<JsonElement, String, JsonArray> converter = converterWrapper(this::jsonAsArray);
+        return getAs(field, converter);
+    }
+
+    /**
+     * Tries to read a custom field from the GraphQL JSON response. The method throws a schema violation
+     * exception if the field name does not contain the <code>_custom_</code> alias suffix added by the library,
+     * which would indicate that the field does not belong to the GraphQL Schema and was not explicitly requested
+     * as a custom field.
+     * 
+     * @param fieldName The field name.
+     * @param element The JSON element parsed by the JSON deserialiser.
+     * @throws SchemaViolationError If the field name does not contain the <code>_custom_</code> suffix.
+     */
+    protected void readCustomField(String fieldName, JsonElement element) throws SchemaViolationError {
+        int end = fieldName.indexOf(AbstractQuery.CUSTOM_FIELD_LABEL);
```

Comment:
I think this allows for potentially unwanted scenarios. If I add a custom field that's called `abc_custom_`, it's internally transformed to `abc_custom__custom`, `indexOf` returns the first occurrence and this code will transform it to `abc`.
If possible I would go for `endsWith()` in the check and `fieldname.substring(0, fieldName.length() - 8)` for the removal of the suffix..

Reply:
Good point, I changed the code and also added a unit test to make sure that we do throw a `SchemaViolationError` for an unknown field.